"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.startLogin = void 0;
const api_1 = require("./api");
exports.startLogin = ({ returnUrl, callbackUrl = '', fingerprint = '', user = '', }) => __awaiter(void 0, void 0, void 0, function* () {
    const form = new FormData();
    form.append('fingerprint', fingerprint);
    form.append('callbackUrl', callbackUrl);
    form.append('returnUrl', returnUrl);
    form.append('accountName', api_1.tenant);
    form.append('scope', api_1.tenant);
    form.append('user', user);
    const response = yield fetch(api_1.api.pub.authentication.startlogin, {
        method: 'POST',
        credentials: 'include',
        headers: {
            accept: 'application/json',
        },
        body: form,
    });
    if (response.status > 300) {
        throw new Error('Something went wrong while logging in');
    }
});
